mimir-build-image/Dockerfile

```diff
@@ -5,7 +5,7 @@
 
 FROM registry.k8s.io/kustomize/kustomize:v5.4.3 as kustomize
 FROM alpine/helm:3.14.4 as helm
-FROM golang:1.23.7-bookworm
+FROM golang:1.23.9-bookworm
```

Comment:
I guess we also did the same in main, so maybe we can just take the image built there? Otherwise we'll end up with N build-images.

Reply:
@colega Do you mean for me to find a main revision where mimir-build-image/Dockerfile is based on Go v1.23.9? Not sure what you mean in practice.